code/datums/underwear/undershirt.dm

```diff
@@ -0,0 +1,447 @@
+	name = "Tank Top, Sun Alt"
+	icon_state = "tank_sun_alt"
+
+/datum/category_item/underwear/undershirt/tank_top_sun_alt_fem
+	name = "Tank Top, Sun Alt Feminine"
+	icon_state = "tank_sun_alt_fem"
+
+/datum/category_item/underwear/undershirt/tank_top_tight
+	name = "Tank Top, Tight"
+	icon_state = "tank_tight"
+
+/datum/category_item/underwear/undershirt/tank_top_tight_fem
+	name = "Tank Top, Tight Feminine"
+	icon_state = "tank_tight_fem"
+
+/datum/category_item/underwear/undershirt/halter
+	name = "Halter Top"
+	icon_state = "halter"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/babydoll_lingerie
+	name = "Babydoll Lingerie"
+	icon_state = "babydoll_lingerie"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/shortsleeve_undershirt
+	name = "Shortsleeve Undershirt"
+	icon_state = "shortsleeve"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/shortsleeve_undershirt_fem
+	name = "Shortsleeve Undershirt, Feminine"
+	icon_state = "shortsleeve_fem"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/shortsleeve_undershirt_vneck
+	name = "Shortsleeve Undershirt, V-Neck"
+	icon_state = "shortsleeve_vneck"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/shortsleeve_undershirt_vneck_fem
+	name = "Shortsleeve Undershirt, V-Neck Feminine"
+	icon_state = "shortsleeve_vneck_fem"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/undershirt
+	name = "Undershirt"
+	icon_state = "undershirt"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/undershirt_fem
+	name = "Undershirt, Feminine"
+	icon_state = "undershirt_fem"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/undershirt_heart
+	name = "Undershirt, Heart"
+	icon_state = "heart"
+
+/datum/category_item/underwear/undershirt/undershirt_heart_fem
+	name = "Undershirt, Heart Feminine"
+	icon_state = "heart_fem"
+
+/datum/category_item/underwear/undershirt/undershirt_heart_echelon
+	name = "Undershirt, Heart Echelon"
+	icon_state = "heart_echelon"
+
+/datum/category_item/underwear/undershirt/undershirt_heart_echelon_fem
+	name = "Undershirt, Heart Echelon Feminine"
+	icon_state = "heart_echelon_fem"
+
+/datum/category_item/underwear/undershirt/shirt_love_nt
+	name = "Undershirt, I<3NT"
+	icon_state = "ilovent"
+
+/datum/category_item/underwear/undershirt/shirt_love_nt_fem
+	name = "Undershirt, I<3NT Feminine"
+	icon_state = "ilovent_fem"
+
+/datum/category_item/underwear/undershirt/shirt
+	name = "Shirt"
+	icon_state = "shirt"
+	has_color = TRUE
+
+/datum/category_item/underwear/undershirt/shirt_alien
+	name = "Shirt, Alien"
+	icon_state = "shirt_alien"
+
+/datum/category_item/underwear/undershirt/shirt_band
+	name = "Shirt, Band"
+	icon_state = "shirt_band"
+
+/datum/category_item/underwear/undershirt/shirt_clown
+	name = "Shirt, Clown"
+	icon_state = "shirt_clown"
+
+/datum/category_item/underwear/undershirt/shirt_ian
+	name = "Shirt, Ian"
+	icon_state = "shirt_ian"

```

Comment:
Memey, I'd consider deleting them but it's up to you.

Reply:
I mean, it's just a dog shirt and a colorful shirt.